pkg/network/server_test.go

```diff
@@ -135,17 +137,29 @@ func TestServerRegisterPeer(t *testing.T) {
 
 	startWithCleanup(t, s)
 
-	s.register <- ps[0]
-	require.Eventually(t, func() bool { return 1 == s.PeerCount() }, time.Second, time.Millisecond*10)
-	s.handshake <- ps[0]
+	var wg sync.WaitGroup
+
+	// Register and handshake peers
+	for i, p := range ps {
```

Comment:
From the commit message I see that this commit contains two different fixes for two different problems. Split this commit into two commits and describe each problem separately.